src/_guides/openapi/specification/v3.1/understanding-structure/http-requests.md

```diff
@@ -0,0 +1,105 @@
+---
+title: HTTP Requests
+authors: phil
+excerpt: Define HTTP requests in your OpenAPI to help users know the rules on what to send and how.
+date: 2024-07-09
+---
+
+Any API handling use-cases more advanced that purely fetching data will need to define a HTTP request body. `POST`, `PATCH`, `PUT`, all allow a HTTP client to send a body: often JSON or XML. This allows more information can be sent than just query parameters, which have limits. 
+
+The request body can be used for:
+
+- Creating new resources (e.g.: booking a train ticket)
+- Updating existing resources (e.g.: updating that booking)
+- Uploading files (e.g.: uploading an avatar to a user profile)
+
+## Structure of Request Bodies
+
+In OpenAPI 3.x, the request body is defined using the `requestBody` object. This object allows you to specify:
+
+- The content type (e.g.: `application/json`, `application/xml`).
+- The schema that defines the structure of the request body.
+- Whether the request body is required or optional.
+- Descriptions for these requests to add context to API documentation.
+
+Let's consider the [Train Travel API](https://bump.sh/blog/modern-openapi-petstore-replacement), which allows users to book train tickets. 
+
+### Creating a Resource
+
+When a user wants to book a train ticket, they need to send details like the passenger's name, trip ID, date of travel, and seat preference, which would look a bit like this:
+
+```yaml
+paths:
+  /bookings:
+    post:
+      summary: Book a train ticket
+      description: Endpoint to book a train ticket
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                passenger_name:
+                  type: string
+                  example: "John Doe"
+                trip_id:
+                  type: string
+                  example: "1234"
+                date:
+                  type: string
+                  format: date
+                  example: "2024-08-15"
+                seat_preference:
+                  type: string
+                  enum: [window, aisle, any]
+                  example: "window"
+```
+
+Here the `requestBody` object defines two important properties:
+
+- `required: true` - indicates that the request body is mandatory for this operation.
+  
+- `content` - specifies that the request body should be in `application/json` format with the following `schema`.
+
+The schema defines the structure of the request body, including properties like passenger_name, train_id, `date`, and `seat_preference`. This can be defined inline like this example, or it can use `components` to share an [existing schema](../data-models/schema-and-data-types.md) and reduce repetition. 
```

Comment:
```suggestion
The schema defines the structure of the request body, including properties like `passenger_name`, `train_id`, `date`, and `seat_preference`. This can be defined inline like this example, or it can use `components` to share an [existing schema](../data-models/schema-and-data-types.md) and reduce repetition. 
```